### DISCUSSION TITLE
Page reload on hyperlink usage.

### DISCUSSION THREAD

**Discussion by AWATV** [remarkjs/remark#1323] [Q&A] [answered]
I have this Markdown:

`[Test](/Minecraft)`

After I use it, it reloads the page,

But if I use the [`react-router-dom`](https://www.npmjs.com/package/react-router-dom) **Link** element:

`<Link to="/Minecraft">Test</Link>`

The page switches correctly.
How i can get the same effect using markdown?

**Comment by remcohaszing** [comment]
We need more info. How are you rendering this markdown? Where does it come from?

**Comment by ChristianMurphy** [comment]
Welcome @AWATV! 👋 
Sorry you ran into some confusion.

> After I use it, it reloads the page

Yes, that is how links work.

> How i can get the same effect using markdown?

By setting a renderer for the link tag to use your preferred custom router link.
This will look different depending on which remark libraries you are using, for `react-markdown` as an example use the `components` prop https://github.com/remarkjs/react-markdown#readme

**Comment by AWATV** [comment, marked as answer]
It helps me fix it:
```
<ReactMarkdown 
  components={{
    a: ({node, ...props}) => <Link to={props.href} {...props} />
  }}
>
```

**Reply by ChristianMurphy** [reply]
The mapping for `a` is correct, and what is described above.
You probably don't want to map `em`, emphasis is not a link, rendering it as a link will likely behave in a very unexpected way.

**Reply by AWATV** [reply]
You are right.
